pkg/controller/kibana/initcontainer/prepare_fs.go

```diff
@@ -0,0 +1,109 @@
+	// that creates the keystore and the main Kibana container.
+	// This is needed in order to have in a same directory both the generated configuration and the keystore file which
+	// is created in /usr/share/kibana/config since Kibana 7.9
+	ConfigSharedVolume = volume.SharedVolume{
+		VolumeName:             settings.ConfigVolumeName,
+		InitContainerMountPath: settings.InitContainerConfigVolumeMountPath,
+		ContainerMountPath:     settings.ConfigVolumeMountPath,
+	}
+
+	// PluginsSharedVolume contains the Kibana plugins/ directory
+	PluginsSharedVolume = volume.SharedVolume{
+		// This volume name is the same as the primary container's volume name
+		// so that the init container does not mount the plugins emptydir volume
+		// on top of /usr/share/kibana/plugins.
+		VolumeName:             settings.PluginsVolumeName,
+		InitContainerMountPath: settings.PluginsVolumeInternalMountPath,
+		ContainerMountPath:     settings.PluginsVolumeMountPath,
+	}
+
+	PluginVolumes = volume.SharedVolumeArray{
+		Array: []volume.SharedVolume{
+			PluginsSharedVolume,
+		},
+	}
+
+	// defaultResources are the default request and limits for the init container.
+	defaultResources = corev1.ResourceRequirements{
+		Requests: map[corev1.ResourceName]resource.Quantity{
+			corev1.ResourceMemory: resource.MustParse("50Mi"),
+			corev1.ResourceCPU:    resource.MustParse("0.1"),
+		},
+		Limits: map[corev1.ResourceName]resource.Quantity{
+			// Memory limit should be at least 12582912 when running with CRI-O
+			corev1.ResourceMemory: resource.MustParse("50Mi"),
+			corev1.ResourceCPU:    resource.MustParse("0.1"),
+		},
+	}
+)
+
+// ConfigVolume returns a SecretVolume to hold the Kibana config of the given Kibana resource.
+func ConfigVolume(kb kbv1.Kibana) volume.SecretVolume {
+	return volume.NewSecretVolumeWithMountPath(
+		kbv1.ConfigSecret(kb),
+		settings.InternalConfigVolumeName,
+		settings.InternalConfigVolumeMountPath,
+	)
+}
+
+// NewInitContainer creates an init container to handle kibana configuration and plugins persistence.
+func NewInitContainer(kb kbv1.Kibana, setDefaultSecurityContext bool) (corev1.Container, error) {
+	v, err := version.Parse(kb.Spec.Version)
+	if err != nil {
+		return corev1.Container{}, err // error unlikely and should have been caught during validation
+	}
+	enablePluginsMounts := v.GTE(HardenedSecurityContextSupportedVersion) && setDefaultSecurityContext
+	container := corev1.Container{
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Name:            settings.InitContainerName,
+		Env:             defaults.PodDownwardEnvVars(),
+		Command:         []string{"/usr/bin/env", "bash", "-c", path.Join(settings.ScriptsVolumeMountPath, KibanaInitScriptConfigKey)},
+		VolumeMounts: []corev1.VolumeMount{
+			ConfigSharedVolume.InitContainerVolumeMount(),
+			ConfigVolume(kb).VolumeMount(),
+		},
+		Resources: defaultResources,
+	}
+
+	if enablePluginsMounts {
```

Comment:
nit: we can remove the variable (or move it closer to where it is used)

```suggestion
	if v.GTE(HardenedSecurityContextSupportedVersion) && setDefaultSecurityContext {
		container.VolumeMounts = append(container.VolumeMounts, PluginsSharedVolume.InitContainerVolumeMount())
	}
```

Reply:
I had previously had this var as I used it twice in the that function, but recent changes had moved it into a package, so I just removed it for now.